tensorflow/core/kernels/training_ops.cc

```diff
@@ -1827,6 +1827,14 @@ class ApplyProximalAdagradOp : public OpKernel {
         errors::InvalidArgument("var and grad do not have the same shape",
                                 var.shape().DebugString(), " ",
                                 grad.shape().DebugString()));
+    const DataType lr_dtype = lr.dtype();
+    OP_REQUIRES(ctx,
+        (l1.dtype() == lr_dtype && 
+        l1.dtype() == lr_dtype && 
```

Comment:
I don't have much context here, but presumably this should be l2.dtype(), since you check l1.dtype() on the line above?